src/modalities/checkpointing/checkpointing_execution.py

```diff
@@ -203,3 +203,33 @@ def load_optimizer_checkpoint(self, optimizer: Optimizer, model: FSDP, file_path
         optimizer.load_state_dict(sharded_optimizer_state_dict)
 
         return optimizer
+
+
+class PytorchToDiscCheckpointing(CheckpointingExecution):
+    def __init__(
+        self,
+        global_rank: int,
+    ):
+        """
+        Implementation of checkpointing to disc via Pytorch
+
+        Args:
+            global_rank (int): global rank within the current process group
+        """
+        self.global_rank = global_rank
+
+    def _save_checkpoint(self, model: FSDP, optimizer: Optimizer, global_train_sample_id: int):
+        raise NotImplementedError
+
+    def _delete_checkpoint(self, global_train_sample_id: int):
+        raise NotImplementedError
+
+    def load_model_checkpoint(self, model: nn.Module, file_path: Path) -> nn.Module:
+        if self.global_rank == 0:
```

Comment:
We intended to implement this in the same way it is implemented in FSDPToDiscCheckpointing

Reply:
yes, `FSDPToDiscCheckpointing` requires a distributed environment. 
However, if we fix that checkpoint conversion never uses a torch dist environment then we can get rid of the rank assignments. This would make a lot of sense, because the checkpoints are stored as PyTorch checkpoints, so why the efforts of initialising the distributed environment?

Reply:
For the checkpoint conversion we don't need a distributed environment, that's true. And we don't initialize it, that was already implemented. However, the class PytorchToDiscCheckpointing could be used in contexts other than checkpoint conversion. And then it might lead to confusion, that the global rank check is not done in load_model_checkpoint, as it is for the class FSDPToDiscCheckpointing

Reply:
Done in 24698a005bd98713844f1ee772e142598f394ed2